modules/dag-l1/src/main/scala/org/tessellation/dag/l1/StateChannel.scala

```diff
@@ -0,0 +1,173 @@
+package org.tessellation.dag.l1
+
+import java.security.KeyPair
+
+import cats.effect.Async
+import cats.effect.std.{Queue, Random}
+import cats.syntax.applicative._
+import cats.syntax.applicativeError._
+import cats.syntax.either._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.traverse._
+
+import scala.concurrent.duration.DurationInt
+
+import org.tessellation.dag.block.BlockValidator
+import org.tessellation.dag.domain.block.DAGBlock
+import org.tessellation.dag.l1.domain.block.{BlockService, BlockStorage}
+import org.tessellation.dag.l1.domain.consensus.block.BlockConsensusCell.isReadyForBlockConsensus
+import org.tessellation.dag.l1.domain.consensus.block.BlockConsensusInput.{
+  OwnRoundTrigger,
+  OwnerBlockConsensusInput,
+  PeerBlockConsensusInput
+}
+import org.tessellation.dag.l1.domain.consensus.block.BlockConsensusOutput.FinalBlock
+import org.tessellation.dag.l1.domain.consensus.block.config.ConsensusConfig
+import org.tessellation.dag.l1.domain.consensus.block.http.p2p.clients.BlockConsensusClient
+import org.tessellation.dag.l1.domain.consensus.block.storage.ConsensusStorage
+import org.tessellation.dag.l1.domain.consensus.block.{BlockConsensusCell, BlockConsensusContext, BlockConsensusInput}
+import org.tessellation.dag.l1.domain.transaction.TransactionStorage
+import org.tessellation.kernel.Cell.NullTerminal
+import org.tessellation.kernel.{CellError, Ω}
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.sdk.domain.cluster.storage.ClusterStorage
+import org.tessellation.sdk.domain.gossip.Gossip
+import org.tessellation.sdk.domain.node.NodeStorage
+import org.tessellation.security.SecurityProvider
+import org.tessellation.security.signature.Signed
+
+import eu.timepit.refined.auto.autoUnwrap
+import fs2.{Pipe, Stream}
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+class StateChannel[F[_]: Async: KryoSerializer: SecurityProvider: Random](
+  blockConsensusClient: BlockConsensusClient[F],
+  blockService: BlockService[F],
+  blockStorage: BlockStorage[F],
+  blockValidator: BlockValidator[F],
+  clusterStorage: ClusterStorage[F],
+  consensusConfig: ConsensusConfig,
+  consensusStorage: ConsensusStorage[F],
+  gossip: Gossip[F],
+  keyPair: KeyPair,
+  myId: PeerId,
+  nodeStorage: NodeStorage[F],
+  peerBlockConsensusInputQueue: Queue[F, Signed[PeerBlockConsensusInput]],
+  peerBlockQueue: Queue[F, Signed[DAGBlock]],
+  transactionStorage: TransactionStorage[F]
+) {
+
+  private implicit val logger = Slf4jLogger.getLogger[F]
```

Comment:
why it needs to be implicit?

Reply:
Currently `BlockConsensusCell` uses it, we can think about refactoring `BlockConsensusCell` in another PR.